entry_types/scrolled/package/src/frontend/Entry.js

```diff
@@ -18,13 +18,20 @@ export default withInlineEditingDecorator('EntryDecorator', function Entry(props
   const [scrollTargetSectionIndex, setScrollTargetSectionIndex] = useState(null);
 
   const entryStructure = useEntryStructure();
-
   useSectionChangeEvents(entryStructure, currentSectionIndex);
 
   const setCurrentSectionIndex = useCallback(index => {
     sectionChangeMessagePoster(index);
     setCurrentSectionIndexState(index);
-  }, [setCurrentSectionIndexState]);
+    let sectionCounter = 0;
+    let chapter = entryStructure.find(chapter => {
+      sectionCounter += chapter.sections.length-1;
```

Comment:
**question**: Why `length - 1`?

Doesn't this mean that `sectionCount` will never be greater than 0 if all chapters only have one section? If we change the test above to simulate scrolling to the second section, we could make sure that this part of the logic is covered.

Reply:
Yes you are right, thanks for pointing out. Fixed it now.